### DISCUSSION TITLE
rmlui progress

### DISCUSSION THREAD

**Discussion by WatchTheFort** [beyond-all-reason/RecoilEngine#554] [1. General Discussion]
@badosu Just wanted to check in to see if there has been any progress on integrating rmlui. I see that they just released version 5.0.
https://github.com/mikke89/RmlUi/releases/tag/5.0

**Comment by badosu** [comment]
Thank you @WatchTheFort, not a lot of progress yet. I'm focusing on https://github.com/beyond-all-reason/spring/discussions/549 right now.

It's my next item on the checklist. I really wanted for the build rework to be integrated first [[PR](https://github.com/beyond-all-reason/spring/pull/448)].

Otherwise I'll have some overhead (that I'm dreading) learning how to do the `spring-static-libs` shit or at least just make it work locally and offload the overhead of integrating to someone else.

**Comment by WatchTheFort** [comment]
@badosu Any status updates on this? CSS font and text handling would be a godsend.

**Comment by badosu** [comment]
Sorry, can't give an ETA. Anyone interested in doing it by their own can proceed.

**Comment by badosu** [comment]
Ok, I'll get back to rmlui now that https://github.com/beyond-all-reason/spring/discussions/549 is resolved, thanks for the patience

**Comment by WatchTheFort** [comment]
@badosu Any progress to report?

**Reply by badosu** [reply]
No, I'm not an active contributor anymore. You can consult the people who are concerned with this now, thank you.

**Reply by badosu** [reply]
Basically this is hard blocked by the rigidity and brittleness of our build system.

Someone can make all the necessary hoops to make the distribution and linking of rmlui with the current system, but personally I don't have the time or ability for it.

See https://github.com/beyond-all-reason/spring/pull/440

**Comment by badosu** [comment]
RmlUI support has been implemented since then, thanks @loveridge @ChrisFloofyKitsune @p2004a
